backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunServiceImpl.java

```diff
@@ -80,10 +82,25 @@ public void getExperimentRunById(
       GetExperimentRunById request,
       StreamObserver<GetExperimentRunById.Response> responseObserver) {
     try {
-      final var response =
-          futureExperimentRunDAO
-              .findExperimentRuns(
-                  FindExperimentRuns.newBuilder().addExperimentRunIds(request.getId()).build())
+      final var requestValidationFuture =
+          InternalFuture.runAsync(
+              () -> {
+                if (request.getId().isEmpty()) {
+                  String errorMessage =
+                      "ExperimentRun ID not found in GetExperimentRunById request";
```

Comment:
```suggestion
                      "ExperimentRun ID not present";
```
The user already knows the type of data they sent, so repeating them back is just a risk for information to be mismatched if we change things.

Reply:
Fixed